x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/literal/Intervals.java

```diff
@@ -417,7 +418,8 @@ public static TemporalAmount parseInterval(Source source, String value, DataType
 
         entries.add(new Entry(IntervalDayTime.class, IntervalDayTime.NAME, IntervalDayTime::new));
         entries.add(new Entry(IntervalYearMonth.class, IntervalYearMonth.NAME, IntervalYearMonth::new));
+        entries.add(new Entry(GeoShape.class, GeoShape.NAME, GeoShape::new));
```

Comment:
Since the new literal is a non interval, the `getNamedWriteables` could be extracted to a separate class `Literals`.

Reply:
Makes sense. I think I will extract this in a separate PR that will go into master and 7.x first. And then resolve conflicts in my branch before merging it. Otherwise, I will be fighting this refactoring until I merge geosql.

Reply:
I opened #40058